src/executors/protect_executor.rs

```diff
@@ -30,11 +39,41 @@ impl<M, N> Executor<SubmitTxToMempool> for ProtectExecutor<M, N>
 where
     M: Middleware,
     M::Error: 'static,
-    N: Middleware,
+    N: Middleware + 'static,
     N::Error: 'static,
 {
     /// Send a transaction to the mempool.
     async fn execute(&self, mut action: SubmitTxToMempool) -> Result<()> {
+        // Acquire a key from the key store
+        let public_address = self
+            .key_store
+            .acquire_key()
+            .await
+            .expect("Failed to acquire key");
```

Comment:
should we loop until a key is acquired? or does the executor automatically retries the same action?